### ISSUE TITLE
MvxCachedImageView error loading in a listview

### ISSUE THREAD

**Issue by Nick1611** [luberda-molinet/FFImageLoading#987]
### Description

I have a MvxCachedImageView in a listview. There is an intermittent error in loading picture that result in loading the ErrorPlaceholder. It could happen:
1) in a list of 10 rows, starting from the middle, say the 7th row
2) after bind the listview to a new data source. the 1st row may get the error immediately

This is the code of the image view
```
<?xml version="1.0" encoding="utf-8"?>
<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:local="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:orientation="horizontal">

  <ffimageloading.cross.MvxCachedImageView
      android:id="@+id/mvxCachedImageView_picture"
      android:layout_width="@dimen/listitem_route_picture_image_size"
      android:layout_height="@dimen/listitem_route_picture_image_size"
      android:layout_marginBottom="4dp"
      android:layout_marginStart="8dp"
      android:layout_marginTop="4dp"
      local:MvxBind="LoadingPlaceholderImagePath 'res:ic_image_grey_700_48dp.png'; ErrorPlaceholderImagePath 'res:ic_error_outline_grey_700_48dp.png'; Transformations Transformations; DownsampleWidth DownsampleWidth; DownsampleUseDipUnits true; ImagePath PicturePath"
      local:layout_constraintStart_toStartOf="parent"
      local:layout_constraintTop_toTopOf="parent" />

```
Pictures are 100% ok as they can be shown in the same listview sometimes without any error. 

From the debug output, there is an error 

```
W/BitmapFactory( 8153): bitmap marked for reuse (2433600 bytes) can't fit new bitmap (3244800 bytes)
03-25 20:34:56.703 D/skia    ( 8153): --- decoder->decode returned false
```
The debug output is attached.
[ffimageloading error log 2.txt](https://github.com/luberda-molinet/FFImageLoading/files/1845487/ffimageloading.error.log.2.txt)

Is there any other parameter that can be used to solve it?

Thanks,

Nick

### Basic Information

- Version with issue: 
- Last known good version: 
- Platform:

**Comment by daniel-luberda** [comment]
Which version? Did you try 2.3.5?

**Comment by Nick1611** [comment]
Yes, it is the latest version 2.3.5 (same with the last version too).

**Comment by daniel-luberda** [comment]
@Nick1611  Could you make a simple repo? I cannot reproduce it.

**Comment by Nick1611** [comment]
Hi Daniel,

 

Thanks for your help. 

 

From the error message, it looks more related to the size of pictures being loaded and the sequence of loading them. I used the same set of pictures to test in different environments and have the following findings:

 

No problem on genymotion (android 5.0 & 7.1) and another Asus old phone with android 4.4

Problem occurs on Samsung S5 (android 6.0.1) and Letv 1s (android 6.0)

 

This seems to be on real phone only, but not emulator.

 

I tested it with 2 set of pictures. Set 1 includes 8 picture (IMG_20180323*). Set 2 with 3 pictures (IMG_20180325*).

Problems occurs randomly when:

1.	Load set 1, loading error starting from the middle of the list
2.	Load set 2. 
3.	Load set 1, then load set 2. Loading error occurs quite often.

 

I put the pictures in the link below.

https://drive.google.com/open?id=1YzRAgFzEitshcbOEklUAKz-qdEXQlpgP


Please let me know if there is any other information I can provide.

 

Thanks,

 

Nick

**Comment by daniel-luberda** [comment]
@Nick1611  Thanks, those images look quite large, it may be related to out of memory issues.

Could you give me:

- downsample parameters values
- transformations used for those images 

?

**Comment by Nick1611** [comment]
These are the parameters:

 

        public double DownsampleWidth => 200d;

 

        public List<ITransformation> Transformations => new List<ITransformation> { new RoundedTransformation(15d, 1d, 1d, 1d, "#000000") };

**Comment by Nick1611** [comment]
I just tried changing the downsample size to 100. The chance of getting the error is less, but still happening.

**Comment by Nick1611** [comment]
After more tests, I would like to give you some update that may help fixing the issue.

My app has 3 places using the image loader for the same set of pictures:

1.	First recycler mentioned in this issue that shows a few rows with MvxCachedImageView on one screen. Sample size is 200dp 
2.	Second recycler showing one full screen image with MvxCachedImageView. Sample size is equal to screen pixel, could be over 1000 pixel
3.	In code using the loader to down sample pictures into 20dp.

As mentioned in earlier posts, the first recycler has intermittent loading error. It happens less if sample size is reduced to 100dp.

After switched to second recycler, though showing & loading one picture at a time, there is no issue at all. Sample size is much higher. Switched back to first recycler, first recycler almost must have the loading error on some / all rows. Continue to use code to downsample picture will have loading error as well.
From the observation, after loading large picture, the state of the loader will become unstable for all subsequent operations. 

Interestingly, the second full screen recycler has no issue! The difference with the first recycler is that second one does not have transformation. After removed transformation from the first recycler, no loading error any more so far.

Once again, these are the setting of the transformation in the firs recycler MvxCachedImageView:

```
local:MvxBind="LoadingPlaceholderImagePath 'res:ic_image_grey_700_48dp.png'; ErrorPlaceholderImagePath 'res:ic_error_outline_grey_700_48dp.png'; Transformations Transformations; DownsampleWidth DownsampleWidth; DownsampleUseDipUnits true; ImagePath PicturePath"

public List<ITransformation> Transformations => new List<ITransformation> { new RoundedTransformation(15d, 1d, 1d, 1d, "#000000") };

```
I am not sure is that something wrong with my transformation setting that is causing the error. At the meantime, I will live without transformation, just not as nice as it could be.

Thanks again,

Nick

**Comment by daniel-luberda** [comment]
That issue should be fixed in 2.3.6 with: https://github.com/luberda-molinet/FFImageLoading/commit/1a650f2086fcf7d10e350cab9b3602601f9fe809

Do you still experience it?

**Comment by Nick1611** [comment]
Hi Daniel,

With the latest version, it is still happening but the frequency is lower. 

As mentioned in my last post, I have 2 recyclers where the second one is full screen. I found that if the second one downsample rate is not large enough, i.e. loading with large memory, the recycler is fine but all subsequent operations of the loader will become unstable. It includes after returned to the first recycler without transformation, and code loading thumbnail. After specified a much lower sample size, it is smooth.

Thanks,

Nick
